pretix_eth/management/commands/confirm_payments.py

```diff
@@ -0,0 +1,247 @@
+                hsh_repr = to_hex(res.hash)
+                val_repr = str(res.value)
+                result_repr = f'{typ_repr}(hash={hsh_repr}, value={val_repr})'
+
+                logger.info(f'Attempting to confirm payment from {result_repr}:')
+
+                try:
+                    payment_id = payment_id_from_wei(res.value)
+                except ValueError as e:
+                    logger.warning(f'  * {e.args[0]}')
+                    logger.warning(f'  * Skipping')
+                    continue
+
+                logger.info(f'  * Payment id is {payment_id}')
+
+                try:
+                    order_payment = OrderPayment.objects.get(id=payment_id)
+                except OrderPayment.DoesNotExist:
+                    logger.warning(f"  * Couldn't find order payment with id {payment_id}")
+                    logger.warning(f'  * Skipping')
+                    continue
+
+                info = order_payment.info_data
+                currency_type = info['currency_type']
+                amount = info['amount']
+
+                if currency_type != expected_currency_type:
+                    logger.warning(f'  * Order payment {order_payment.id} was expecting payment in {currency_type}!!!')  # noqa: E501
+                    logger.warning(f'  * Given payment was in {expected_currency_type}')
+                    logger.warning(f'  * Skipping')
+                    continue
+
+                if res.value < amount:
+                    logger.warning(f'  * Order payment {order_payment.id} was expecting payment of at least {amount}!!!')  # noqa: E501
+                    logger.warning(f'  * Given payment was only for {res.value}')
+                    logger.warning(f'  * Skipping')
+                    continue
+
+                if dry_run:
+                    logger.info(f'  * DRY RUN: Would confirm order payment {order_payment.id}')
+                else:
+                    logger.info(f'  * Confirming order payment {order_payment.id}')
+                    order_payment.confirm()
+
+    def handle(self, *args, **options):
+        dry_run = options['dry_run']
+        start_block = options['start_block']
+        end_block = options['start_block']
+        token_address = options['token_address']
+
+        # Instantiate transaction provider based on chosen API
+        api = options['api']
+        txn_provider_class = TXN_PROVIDERS[api]
+        if api.startswith('etherscan-'):
+            try:
+                api_key = os.environ['ETHERSCAN_API_KEY']
+            except KeyError:
+                raise ValueError('Etherscan api key not found in "ETHERSCAN_API_KEY" env var')
+            txn_provider = txn_provider_class(api_key=api_key)
+        else:
+            txn_provider = txn_provider_class()
+
+        # Determine wallet address
+        event_slug = options['event_slug']
+        if event_slug is not None:
+            payment_provider = Ethereum(Event.objects.get(slug=event_slug))
```

Comment:
Maybe catch the exception that would be raised here if no `Event` is found with the given slug and issue a nice error message prior to exiting?